### ISSUE TITLE
Adapter for styled-components

### ISSUE THREAD

**Issue by pupudu** [chakra-ui/chakra-ui#2338]
# Feature request

## Is your feature request related to a problem? Please describe.
No

## Describe the solution you'd like
N/A

## Describe alternatives you've considered
N/A

## Additional context
First of all, great library 👏 Probably the best one out there. Sugoii

I work in a company which has a frontend codebase invested heavily on styled-components with server side rendering. We are considering using chakra-ui as the foundation for the next iteration of our design system. But unfortunately we are getting a lot of push back since there will be two style engines running in parallel if we do use chakra.

I looked at the codebase and it seems it's really modular, and I could **fork and replace emotion with styled-components**. That is one possibility for us. But that would make staying up-to-date with upstream changes hard. 

I was wondering if you guys will be interested in a generic adapter like solution which will let people like me to switch to styled-components with a little bit of extra code. For instance, Smooth-UI supports this: https://smooth-ui.smooth-code.com/

Happy to give it a crack and make a PR. Needed to see if you guys are interested in that first of all.

Ariagatou Gozaimasu 🙏

**Comment by segunadebayo** [comment]
Thank you for this amazing remark @pupudu.

We gave this some consideration and have decided not to provide an adapter for styled-components for now to reduce the complexity of our codebase.

We might consider this in the future though. Please feel free to fork and replace emotion with styled-components as needed. If you ever, publish it, you can also leave a link to the package here as well so others can find it.

Cheers.

**Comment by pupudu** [comment]
Cheers mate 👍

**Comment by zebapy** [comment]
@segunadebayo  is there consideration for switching to styled-components entirely? I'm in process of trying to port from SC to emotion so we can adopt chakra, but emotion just seems to be a bit behind and SC is generally taking over the space more I think.

**Comment by pupudu** [comment]
> @segunadebayo is there consideration for switching to styled-components entirely? I'm in process of trying to port from SC to emotion so we can adopt chakra, but emotion just seems to be a bit behind and SC is generally taking over the space more I think.

Hey. I'm the OP of this issue. I ended up using mui instead of chakra because mui supports a SC adaptor. However, I have to disagree that SC is better than emotion. I've realised that the emotion is actually more flexible and stable specially with SSR. There are some notes in mui documentation regarding the issues with styled components. SC certainly has more stars and maybe a bigger community, but that's mostly because it has always been so.
